.github/workflows/test-build-cmake-presets.yml

```diff
@@ -0,0 +1,101 @@
+name: WolfBoot CMake Presets Build
+
+on:
+  push:
+    # TODO: branches: [ 'master', 'main', 'release/**' ]
+    branches: [ '*' ]
+  pull_request:
+    branches: [ "*" ]
+
+permissions:
+  contents: read
+
+jobs:
+  ubuntu-cmake:
+    name: Build on Ubuntu
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    defaults:
+      run:
+        shell: bash
+
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - stm32l4
+          - stm32h7
+          - stm32g0
+        include:
+          # Optional per-target cache variables you might want to pass later.
+          # Keep empty for now to avoid guessing addresses.
+          - target: stm32l4
+            extra_cache: ""
+          - target: stm32h7
+            extra_cache: ""
+          - target: stm32g0
+            extra_cache: ""
+
+    steps:
+      - name: Checkout (with submodules)
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # Lock down network/runner
+      # See https://github.com/step-security/harden-runner/releases
+      # Currently only supported on Ubuntu
+
+      # ARM GCC toolchain (adds the bin dir to PATH)
+      - name: Set up ARM none-eabi GCC 14.x
+        uses: carlosperate/arm-none-eabi-gcc-action@v1
```

Comment:
@danielinux  I'd like to specifically point out this one for review. 

How to ensure this is vetted and appropriate? It seems well established. Still (?)

We may consider also doing something like this as noted:

https://github.com/step-security/harden-runner/blob/main/README.md